Classes/common/HTTP/CDTURLSession.m

```diff
@@ -45,10 +48,21 @@ - (instancetype)initWithDelegate:(NSObject<NSURLSessionDelegate> *)delegate
         _thread = thread;
         _interceptors = [NSArray arrayWithArray:requestInterceptors];
 
-        NSURLSessionConfiguration *config = [NSURLSessionConfiguration defaultSessionConfiguration];
+        NSURLSessionConfiguration *config;
+        // Create a unique session id using the address of self.
+        NSString *sessionId = [NSString stringWithFormat:@"com.cloudant.sync.sessionid.%p", self];
+        if ([[NSURLSessionConfiguration class] respondsToSelector:@selector(backgroundSessionConfigurationWithIdentifier:)]) {
+            config = [NSURLSessionConfiguration backgroundSessionConfigurationWithIdentifier:sessionId];
+        } else {
+            config = [NSURLSessionConfiguration backgroundSessionConfiguration:sessionId];
+        }
+        config = [sessionConfigDelegate customiseNSURLSessionConfiguration:config];
+
         _session = [NSURLSession sessionWithConfiguration:config
-                                                 delegate:delegate
+                                                 delegate:self
                                             delegateQueue:nil];
+
+        _taskMap = [NSMapTable strongToWeakObjectsMapTable];
```

Comment:
The way `_taskMap` works could probably do with some documentation. Does this type of `NSMapTable` mean that values are removed if keys are reclaimed through reference counting?